e2etest/scenario_helpers.go

```diff
@@ -541,11 +541,82 @@ func (scenarioHelper) generateAzureFilesFromList(c asserter, shareURL azfile.Sha
 }
 
 func (s scenarioHelper) enumerateShareFileProperties(a asserter, shareURL azfile.ShareURL) map[string]*objectProperties {
+	var dirQ []azfile.DirectoryURL
+	result := make(map[string]*objectProperties)
+
+	root := shareURL.NewRootDirectoryURL()
+	dirQ = append(dirQ, root)
+	for i := 0; i < len(dirQ); i++ {
+		currentDirURL := dirQ[i]
+		for marker := (azfile.Marker{}); marker.NotDone(); {
+			lResp, err := currentDirURL.ListFilesAndDirectoriesSegment(context.TODO(), marker, azfile.ListFilesAndDirectoriesOptions{})
+			a.AssertNoErr(err)
+
+			// Process the files and folders we listed
+			for _, fileInfo := range lResp.FileItems {
+				fileURL := currentDirURL.NewFileURL(fileInfo.Name)
+				fgpr, err := fileURL.GetProperties(context.TODO())
```

Comment:
Minor: in terms of coding style, fgpr is not a "Go-ish" name. Idiomatic Go code would probably just use "p" as the name of this variable.  But, here in AzCopy, we tend to be a bit more verbose.  So "props" might be a good AzCopy idiomatic name.  

BTW You'll often also see us use two-word names, such as fileSize.  I think it could be argued that our two word names are too long by Go standards in some cases (e.g. when the meaning would be obvious with one word) but justifiable in some others (IMHO when they help with readability of non-obvious code).

But one thing that you would never see in AzCopy or in conventional idiomatic Go with shorter names, would be something like fgpr.  So I'd recommend avoiding names of that style.